backend/src/test/java/ca/bc/gov/backendstartapi/endpoint/FavouriteActivityEndpointTest.java

```diff
@@ -0,0 +1,261 @@
+
+    mockMvc
+        .perform(
+            post(API_PATH)
+                .with(csrf().asHeader())
+                .header(CONTENT_HEADER, JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(stringifyCreate(ActivityEnum.PARENT_TREE_ORCHARD)))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.activity").value(ActivityEnum.PARENT_TREE_ORCHARD.toString()))
+        .andExpect(jsonPath("$.highlighted").value("false"))
+        .andExpect(jsonPath("$.enabled").value("true"))
+        .andReturn();
+
+    FavouriteActivityEntity activityUpdated = activityEntity.withHighlighted(true);
+
+    when(favouriteActivityService.updateUserActivity(any(), any())).thenReturn(activityUpdated);
+
+    mockMvc
+        .perform(
+            put(API_PATH + "/{id}", activityEntity.getId())
+                .with(csrf().asHeader())
+                .header(CONTENT_HEADER, JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(stringifyUpdate(true, true)))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.activity").value(ActivityEnum.PARENT_TREE_ORCHARD.toString()))
+        .andExpect(jsonPath("$.highlighted").value("true"))
+        .andExpect(jsonPath("$.enabled").value("true"))
+        .andReturn();
+  }
+
+  @Test
+  @DisplayName("deleteUserFavoriteActivity")
+  @WithMockUser(roles = "user_write")
+  void deleteUserFavoriteActivity() throws Exception {
+    FavouriteActivityEntity activityEntity = createEntity(ActivityEnum.PARENT_TREE_ORCHARD);
+    activityEntity.setId(10000L);
+
+    when(favouriteActivityService.createUserActivity(any())).thenReturn(activityEntity);
+
+    mockMvc
+        .perform(
+            post(API_PATH)
+                .with(csrf().asHeader())
+                .header(CONTENT_HEADER, JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(stringifyCreate(ActivityEnum.PARENT_TREE_ORCHARD)))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.activity").value(ActivityEnum.PARENT_TREE_ORCHARD.toString()))
+        .andExpect(jsonPath("$.highlighted").value("false"))
+        .andExpect(jsonPath("$.enabled").value("true"))
+        .andReturn();
+
+    FavouriteActivityEntity activityUpdated = activityEntity.withHighlighted(true);
+
+    doNothing().when(favouriteActivityService).deleteUserActivity(any());
+
+    mockMvc
+        .perform(
+            delete(API_PATH + "/{id}", activityUpdated.getId())
+                .with(csrf().asHeader())
+                .header(CONTENT_HEADER, JSON)
+                .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andReturn();
+  }
+
+  @Test
+  @DisplayName("createFavoriteActivityEndpointTest")
+  void createFavoriteActivityEndpointTest() {
+    FavouriteActivityEndpoint activityEndpoint = new FavouriteActivityEndpoint();
+    activityEndpoint.setFavouriteActivityService(favouriteActivityService);
+  }
```

Comment:
The SonarLint plugin on IntelliJ pointed that no assertion is made in this test, which I believe is a fair complaint, but there isn't much to test here besides it not exploding for unknown reasons.

If this is here for code coverage only, [we can apparently annotate methods we don't want to be included in code coverage](https://www.baeldung.com/jacoco-report-exclude#bd-excluding-with-custom-annotation), although I'm not thrilled with the naming requirement.

Once we find a good name for such annotation we can use it for boilerplate methods such as this constructor, but we don't need that right now.

Reply:
All right. I'll remove this test case then.